### PR TITLE
Js fixes 2

### PR DESCRIPTION
**Pull request by dbMundada**
- [Use pattern from Collection Express rule](https://github.com/Privado-Inc/privado-core/commit/804ba1e303ca2b26f3a6cda6eceaa9bb23dcae71)
- [Remove FPs from import/require statements](https://github.com/Privado-Inc/privado-core/commit/b694bef93626879c954270783fea704208a789cd)
- [Add support for parsing block with method, path & handler for Collection](https://github.com/Privado-Inc/privado-core/commit/cdcb58cd9fb16544b676809beeeb3248c32e96cf)

### CODE REVIEW

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala [hunk 1]
```diff
@@ -32,7 +33,8 @@ class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCp
     // TODO: Need to support more frameworks Hapijs, Koa, Loopback, Sails, Restify, Connect, AdonisJS
     val EXPRESS_CLIENT_PATTERN =
       "(?:express|fetch|@feathersjs/feathers|fastify|restify|@nestjs/cli|itty-router|koa-router|@ioc[:]Adonis|@adonisjs|@sails|sails|.*loopback).*"
-    val expressCollectionCalls = cpg.call.methodFullName(EXPRESS_CLIENT_PATTERN).l
+    val expressCollectionCalls =
+      cpg.call.methodFullName(collectionRuleInfo.patterns.headOption.getOrElse(EXPRESS_CLIENT_PATTERN)).l
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala:37]: I would prefer the idea of not using hardCoded `Express_client_pattern`, If this pattern is not present in rules, this logic should not get executed

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala [hunk 2]
```diff
@@ -91,17 +95,21 @@ class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCp
     val objectKeyValuePairs      = block.astChildren.isCall.name("<operator>.assignment").l
     var handlerMethod: MethodRef = null
     var path: String             = ""
-    for (keyVal <- objectKeyValuePairs) {
-      if (keyVal.astChildren.nonEmpty) {
-        val objKeyVal = keyVal.astChildren
-        val key       = objKeyVal.isCall.head
+    if (objectKeyValuePairs.nonEmpty && objectKeyValuePairs.isInstanceOf[List[Call]]) {
+      for (keyVal <- objectKeyValuePairs) {
+        if (keyVal.astChildren.nonEmpty && keyVal.astChildren.isInstanceOf[Traversal[AstNode]]) {
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala:100]: is this check `keyVal.astChildren.isInstanceOf[Traversal[AstNode]]` needed as `astChildren` will return AstNode as per my understanding

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala [hunk 3]
```diff
@@ -91,17 +95,21 @@ class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCp
...
+          val objKeyVal = keyVal.astChildren
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala:101]: is this object `objKeyVal` used anywhere, if not you can say 
`val key = keyVal.astChildren.isCall.l`

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala [hunk 4]
```diff
@@ -91,17 +95,21 @@ class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCp
...
+          val key       = objKeyVal.isCall
 
-        if (key.code.contains(pathField) && key.astSiblings.head.isLiteral) {
-          path = key.astSiblings.head.code
-        }
+          if (key.nonEmpty) {
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala:104]: is the code inside is getting executed as `key` is a traversal/iterator, which can only be used once?

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/source/IdentifierTagger.scala [hunk 5]
```diff
@@ -124,8 +124,10 @@ class IdentifierTagger(cpg: Cpg, ruleCache: RuleCache, taggerCache: TaggerCache)
             // typeDeclVal: models.py:<module>.Profile
             val impactedObjects = cpg.identifier
               .filter(n => typeDeclVal.endsWith(n.typeFullName))
-//              _.astSiblings failing in some cases to removing this filter for now.
-//              .whereNot(_.astSiblings.isCall.name("import"))
+              .filter(n => {
+                if (n.contains("astSiblings")) { n.astSiblings.isCall.name("import|require").isEmpty }
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/languageEngine/javascript/tagger/source/IdentifierTagger.scala:128]: Do we need this check?

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala (hunk 1 above)
Comment (dbMundada) [reply src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala]: Updated logic

File: src/main/scala/ai/privado/languageEngine/javascript/tagger/source/IdentifierTagger.scala (hunk 5 above)
Comment (dbMundada) [reply src/main/scala/ai/privado/languageEngine/javascript/tagger/source/IdentifierTagger.scala:128]: Yes, else scan for some repos is breaking.

Comment (khemrajrathore) [reply src/main/scala/ai/privado/languageEngine/javascript/tagger/source/IdentifierTagger.scala:128]: Interesting, can you try with `n.start.astSiblings` that should work

Comment (dbMundada) [reply src/main/scala/ai/privado/languageEngine/javascript/tagger/source/IdentifierTagger.scala:128]: tried  not working.

**Review by khemrajrathore — APPROVED** [review]
approve

**Pull request merged**